pkg/types/powervs/machinepools.go

```diff
@@ -0,0 +1,55 @@
+package powervs
+
+// +kubebuilder:validation:Enum=capped;dedicated;shared
+type procType string
+
+// MachinePool stores the configuration for a machine pool installed on IBM Power VS.
+type MachinePool struct {
+	// VolumeIDs is the list of volumes attached to the instance.
+	//
+	// +optional
+	VolumeIDs []string `json:"volumeIDs,omitempty"`
+
+	// Memory defines the memory in GB for the instance.
+	//
+	// +optional
+	Memory string `json:"memory"`
+
+	// Processors defines the processing units for the instance.
+	//
+	// +optional
+	Processors string `json:"processors"`
```

Comment:
All optional values should be `omitempty`.
```suggestion
	Processors string `json:"processors,omitempty"`
```